lib/cloudregister/registerutils.py

```diff
@@ -530,73 +530,80 @@ def setup_registry(registry_fqdn, username, password):
 
 
 # ----------------------------------------------------------------------------
-def get_registry_credentials():
+def get_registry_credentials(set_new):
     """Read the registry credentials file
        and return its content or an empty dict."""
     config_json = {}
-    failed = False
-    if os.path.exists(REGISTRY_CREDENTIALS_PATH):
-        file_error = False
-        try:
-            with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
-                config_json = json.load(cred_json)
-        except OSError as error:
-            logging.info(
-                'Unable to open %s: %s, preserving file as %s.bak, '
-                'writing new credentials' % (
-                    REGISTRY_CREDENTIALS_PATH, error, REGISTRY_CREDENTIALS_PATH
-                )
-            )
-            file_error = True
-        except json.decoder.JSONDecodeError:
-            logging.info(
-                'Unable to parse existing %s, preserving file as %s.bak, '
-                'writing new credentials' % (
-                    REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
-                )
-            )
-            file_error = True
+    failed = None
+    if not os.path.exists(REGISTRY_CREDENTIALS_PATH):
+        logging.info('Credentials file does not exist. Nothing to do')
+        return config_json, failed
```

Comment:
The function is called `get_registry_credentials()` in this case we cannot return the credentials, so I would consider that a failure, but we return `None` maybe indicating no failure? The intend here is not clear and we need to sort out what it should be.

Reply:
Yes, the returning of `None` is intentional to differentiate the case of the non existing file with the case of existing file and failing and failing to open it or accessing the file and being empty, for example

Reply:
Moved the check outside the method, I think that makes the intent clear